### PR TITLE
feat: introducing boot changes to support Kratos flow

### PR DESCRIPTION
**Pull request by rebelchris**
Added optional flow to support the Kratos users.
Tried to use as much as the original flow as we could.

Once we reach 90% user base we can modify the flow and remove all legacy code.

Things to note:
- Does Kratos whoami refresh the token?
- We should introduce roles in the Kratos traits if possible (it's only moderator rights)

WT-296

### CODE REVIEW

File: src/cookies.js [hunk 1]
```diff
@@ -7,17 +7,17 @@ const extractDomain = (ctx) => {
   while (parts.length > 2) {
     parts.shift();
   }
-  return parts.join('.');
+  return '127.0.0.1';
```
Comment (rebelchris) [inline src/cookies.js:10]: We should fix this 😅

File: .env [hunk 2]
```diff
@@ -24,6 +24,8 @@ SENDGRID_API_KEY=''
 
 PRIMARY_AUTH_ORIGIN='http://localhost:4000'
 
+KRATOS_ORIGIN='http://127.0.0.1:4433'
```
Comment (idoshamun) [inline .env:27]: Do you think we need to replace all `localhost` with `127.0.0.1` to make it consistent?

File: src/auth.js [hunk 3]
```diff
@@ -26,3 +27,32 @@ export const validateRefreshToken = async (ctx) => {
   }
   return shouldRefreshToken;
 };
+
+const validateKratosToken = async (ctx) => {
+  try {
+    const res = await rp(`${config.kratosOrigin}/sessions/whoami`, { headers: ctx.req.headers });
+    const kratos = JSON.parse(res);
+    if (kratos?.identity?.id) {
+      ctx.state.user = { userId: kratos?.identity?.id };
```
Comment (idoshamun) [inline src/auth.js:36]:
```suggestion
      ctx.state.user = { userId: kratos.identity.id, isKratos: true };
```

File: src/auth.js [hunk 4]
```diff
@@ -26,3 +27,32 @@ export const validateRefreshToken = async (ctx) => {
...
+    throw e;
+  }
+
+  return false;
+};
+
+export const validateToken = async (ctx) => {
+  const isKratos = ctx.cookies.get(config.cookies.kratos.key);
+  if (isKratos) {
+    ctx.state.isKratos = true;
```
Comment (idoshamun) [inline src/auth.js:53]: I suggest moving it into `validateKratosToken`. (suggestion provided already above)

**Review by idoshamun — CHANGES_REQUESTED** [review]
Very well thought PR!
I love the backward compatibility mindset you demonstrated here. 🤩

File: .env (hunk 2 above)
Comment (rebelchris) [reply .env:27]: Probably nicer, no urgent need though.
I'll do it for the sake of uniforming it 🙏

**Review by idoshamun — APPROVED** [review]

**Pull request merged**